### PR TITLE
Correct BARINIT macro for pthread_barrier_wait

### LINKED ISSUE #2: BARINIT incomplete for some applications
Some applications omit the count parameter for BARINIT. While this works for the default macro (`pthread.m4.stougie`), it does not work when one wishes to use `pthread_barrier_init`, which requires a count to be passed.

As an example, inside codes/apps/barnes/code.c.in:

```
-   BARINIT(Global->Barload);
-   BARINIT(Global->Bartree);
-   BARINIT(Global->Barcom);
-   BARINIT(Global->Baraccel);
-   BARINIT(Global->Barstart);
-   BARINIT(Global->Barpos);
+   BARINIT(Global->Barload, NPROC);
+   BARINIT(Global->Bartree, NPROC);
+   BARINIT(Global->Barcom, NPROC);
+   BARINIT(Global->Baraccel, NPROC);
+   BARINIT(Global->Barstart, NPROC);
+   BARINIT(Global->Barpos, NPROC);
```

### PR DESCRIPTION
**Pull request by mariobadr**
Fixes #2 and updates the link to the Splash-3 paper found in IEEE Xplore.

### CODE REVIEW

File: codes/pthread_macros/pthread.m4.POSIX_BARRIER [hunk 1]
```diff
@@ -0,0 +1,98 @@
+dnl Found on the web.  Written by Bastiaan Stougie, October 2003
+dnl Modified by Alberto Ros and Christos Sakalis, 2015
```
Comment (SakalisC) [inline codes/pthread_macros/pthread.m4.POSIX_BARRIER:2]: Would you like to add your name here maybe? It's not necessary if you don't want to, but I thought I should ask anyway.

**Review by SakalisC — COMMENTED** [review]
Thanks for the patch! Everything looks good, my only comment is if you would like to have the new macros attributed to you. It's not a problem if you don't, I will merge it as is in that case.

File: codes/pthread_macros/pthread.m4.POSIX_BARRIER (hunk 1 above)
Comment (mariobadr) [reply codes/pthread_macros/pthread.m4.POSIX_BARRIER:2]: That's ok, no need for attribution. I simply used the same thing found in the POSIX_BARRIER file from the null macros,

Comment (SakalisC) [reply codes/pthread_macros/pthread.m4.POSIX_BARRIER:2]: 👍

**Pull request merged**